File: cxx-mnp-plasmon/examples/spectrum_csv.cpp
#include "../include/mnp_plasmon.hpp"
#include <filesystem>
#include <fstream>

int main() {
    std::filesystem::create_directories("docs/example-data");
    std::ofstream out("docs/example-data/cpp.csv");
    if (!out) {
        return 1;
    }

    out << "wavelength_nm,c_ext,c_sca,c_abs\n";
    for (int wl = 400; wl <= 700; wl += 5) {
        auto r = mnp::MnpPlasmon::simulate_sphere_response("Au", static_cast<double>(wl), 25.0, 1.33);
        out << wl << ',' << r.c_ext << ',' << r.c_sca << ',' << r.c_abs << "\n";
    }

    return 0;
}

File: rust-mnp-plasmon/examples/spectrum_csv.rs
use std::fs::{self, File};
use std::io::Write;

fn main() {
    let mut output = String::from("wavelength_nm,c_ext,c_sca,c_abs\n");

    for wavelength in (400..=700).step_by(5) {
        let response = mnp_plasmon::sphere_response(wavelength as f64, 25.0, "Au", 1.33)
            .expect("sphere response should compute");
        output.push_str(&format!(
            "{},{},{},{}\n",
            wavelength,
            response.c_ext,
            response.c_sca,
            response.c_abs
        ));
    }

    fs::create_dir_all("docs/example-data").expect("create docs/example-data");
    let mut file = File::create("docs/example-data/rust.csv").expect("create rust.csv");
    file.write_all(output.as_bytes()).expect("write csv");
}
